src/WebJobs.Script/Environment/EnvironmentExtensions.cs

```diff
@@ -246,12 +246,24 @@ public static string GetSubscriptionId(this IEnvironment environment)
             return null;
         }
 
+        /// <summary>
+        /// Gets the name of the currently deployed slot name.
+        /// </summary>
         public static string GetRuntimeSiteName(this IEnvironment environment)
         {
             string runtimeSiteName = environment.GetEnvironmentVariable(AzureWebsiteRuntimeSiteName);
             return runtimeSiteName?.ToLowerInvariant();
         }
 
+        /// <summary>
+        /// Gets the name of the app's slot, if it exists.
+        /// </summary>
+        public static string GetSlotName(this IEnvironment environment)
+        {
+            string slotName = environment.GetEnvironmentVariable(AzureWebsiteSlotName);
```

Comment:
I remember being confused by this in the past. We always restart a function app during a swap, right? So this will always be correct and we can cache it? 

@fabiocav do you know?

Reply:
Where do you see this being cached?

Reply:
It's not right now -- but we have a new AppServiceOptions type to hold stuff like this. If we know that it's stable at specialization time, we can put it there.